src/org/omegat/core/data/RealProject.java

```diff
@@ -336,39 +336,47 @@ public synchronized void loadProject(boolean onlineMode) {
             Core.getMainWindow().showStatusMessageRB("CT_LOADING_PROJECT");
 
             if (remoteRepositoryProvider != null) {
-                tmxPrepared = null;
-                glossaryPrepared = null;
-                
-                remoteRepositoryProvider.switchAllToLatest();
+                try {
+                    tmxPrepared = null;
+                    glossaryPrepared = null;
+
+                    remoteRepositoryProvider.switchAllToLatest();
 
-                // Add filters.xml and segmentation.conf
-                for (String file : new String[] {
-                    '/' + config.getProjectInternalRelative() + FilterMaster.FILE_FILTERS,
-                    '/' + config.getProjectInternalRelative() + SRX.CONF_SENTSEG }) {
-                remoteRepositoryProvider.copyFilesFromRepoToProject(file);
+                    // Add filters.xml and segmentation.conf
+                    for (String file : new String[]{
+                            '/' + config.getProjectInternalRelative() + FilterMaster.FILE_FILTERS,
+                            '/' + config.getProjectInternalRelative() + SRX.CONF_SENTSEG}) {
+                        remoteRepositoryProvider.copyFilesFromRepoToProject(file);
+                    }
+                    // After adding filters.xml and segmentation.conf, we must reload them again
+                    config.loadProjectFilters();
+                    config.loadProjectSRX();
+
+                } catch (Exception e) {
```

Comment:
It doesn't seem right that *any* exception in this block leads to being considered offline. We should be defining our own `NetworkUnavailableException` or similar to wrap each remote provider's specific exception, then handle only that one.

Reply:
Should this wrapping be done inside RemoteRepositoryProvider? I see no reason to do it in RealProject.

Reply:
Definitely not in `RealProject`. Whether `RemoteRepositoryProvider` is the best place or not, I don't recall off the top of my head. There might be a lower layer that is more appropriate, depending on the specific provider's implementation.